### PR TITLE
git-remote-gcrypt 1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by srl295**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/git-remote-gcrypt.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class GitRemoteGcrypt < Formula
+  desc "git remote helper for GPG-encrypted remotes"
+  homepage "https://spwhitton.name/tech/code/git-remote-gcrypt/"
+  url "https://git.spwhitton.name/git-remote-gcrypt.git"
```
Comment (bfontaine) [inline Formula/git-remote-gcrypt.rb]: We need a stable release here.

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/git-remote-gcrypt.rb (hunk 1 above)
Comment (srl295) [inline Formula/git-remote-gcrypt.rb]: @bfontaine sorry… I thought the `version 1.0.0` would do that (specify the tag from the repo). Refactored and verified.

File: Formula/git-remote-gcrypt.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  depends_on "docutils" => :build
+
+  def install
+    bin.install "git-remote-gcrypt"
+    system "rst2man.py", "README.rst", "git-remote-gcrypt.1"
+    man1.install "git-remote-gcrypt.1"
+  end
+
+  test do
+    system "#{bin}/git-remote-gcrypt"
```
Comment (MikeMcQuaid) [inline Formula/git-remote-gcrypt.rb]: Use `assert_match "something", shell_output("#{bin}/git-remote-gcrypt")` to match some output here.

File: Formula/git-remote-gcrypt.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  url "https://git.spwhitton.name/git-remote-gcrypt.git",
+      :tag => "1.0.0",
+      :revision => "30b3f5e3356bba404eb630807879b8d4989f8c6e"
+  sha256 "9ee9554a7bc627b374e1f80fab4b5539b62b505613c95fdafdc884341ea2cddf"
+  head "https://git.spwhitton.name/git-remote-gcrypt.git"
+
+  depends_on "docutils" => :build
+
+  def install
+    bin.install "git-remote-gcrypt"
```
Comment (MikeMcQuaid) [inline Formula/git-remote-gcrypt.rb]: Use `install.sh` instead: https://git.spwhitton.name/?p=git-remote-gcrypt.git;a=blob;f=install.sh;h=d84f31ec116721915d2e054faafee67dbf3a0e27;hb=HEAD

File: Formula/git-remote-gcrypt.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+class GitRemoteGcrypt < Formula
+  desc "git remote helper for GPG-encrypted remotes"
+  homepage "https://spwhitton.name/tech/code/git-remote-gcrypt/"
+  url "https://git.spwhitton.name/git-remote-gcrypt.git",
+      :tag => "1.0.0",
+      :revision => "30b3f5e3356bba404eb630807879b8d4989f8c6e"
```
Comment (MikeMcQuaid) [inline Formula/git-remote-gcrypt.rb]: May be worth asking them for a tarball as that may be less expensive for them.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by spwhitton** [comment]
Tarballs available from the Debian mirror network: http://ftp.debian.org/debian/pool/main/g/git-remote-gcrypt/

File: Formula/git-remote-gcrypt.rb (hunk 4 above)
Comment (spwhitton) [inline Formula/git-remote-gcrypt.rb]: Here: http://ftp.debian.org/debian/pool/main/g/git-remote-gcrypt/

**Comment by MikeMcQuaid** [comment]
> Tarballs available from the Debian mirror network: http://ftp.debian.org/debian/pool/main/g/git-remote-gcrypt/

@spwhitton Would you rather we used that to save you bandwidth?

**Comment by spwhitton** [comment]
Yes please :)

Though please keep the homepage pointing at my site.

-- 
Sean Whitton

**Comment by srl295** [comment]
@spwhitton @MikeMcQuaid there's also the github mirror… but `brew audit` rejected it as being "not the main fork"

**Comment by MikeMcQuaid** [comment]
@srl295 The GitHub mirror would be preferable. Ignore the fork warning.
@spwhitton Would you like your fork either detached (so it doesn't show as forked from anything) or the primary node (so e.g. https://github.com/joeyh/git-remote-gcrypt shows as forked from yours)? I'm GitHub staff so can update that accordingly.

**Comment by spwhitton** [comment]
On Fri, Nov 18, 2016 at 08:19:30AM -0800, Mike McQuaid wrote:
> @spwhitton Would you like your fork either detached (so it doesn't show as
> forked from anything) or the primary node (so e.g. https://github.com/joeyh/
> git-remote-gcrypt shows as forked from yours)? I'm GitHub staff so can update
> that accordingly.

Thank you for your offer to update the forking relationships.

The reason I maintain a mirror of the project on GitHub is to make it
easier for people to find it.  Detaching the repository would not serve
that purpose, as then it wouldn't show up in the list of forks.

On the other hand, making it the primary node would make it easier to
find my repository.  It would be completely clear that patches/PRs
should be submitted to me, not to @joeyh or @bluss.

I'm concerned, though, that it might imply that I was the originator of
the project.  I maintain it upstream and in Debian and Ubuntu, but I did
not come up with the idea nor did I write very much of the code.  So it
seems dishonest for my repo to be the primary node.

Have I misunderstood what it means to be the primary node?  I am not
really active on the social side of GitHub, so I might have misconstrued
it.

-- 
Sean Whitton

**Comment by MikeMcQuaid** [comment]
@spwhitton Your understanding there seems to be correct. Given that @joeyh's and @bluss's forks both state they are unmaintained and @joeyh's points to yours explicitly (and I'd argue @bluss's implicitly does) it seems better to me for recognising which project is being actively developed. Ultimately, it's up to you though so if you're OK with me doing so let me know otherwise I'll leave it as is. Thanks for all your work, regardless!

**Comment by spwhitton** [comment]
@MikeMcQuaid Probably best to leave it as it is for now.  Thanks again
for the offer.

-- 
Sean Whitton

**Comment by MikeMcQuaid** [comment]
@spwhitton Sure, thanks.
@srl295 Sounds like there should be enough information for you to update this now. No rush.

File: Formula/git-remote-gcrypt.rb (hunk 3 above)
Comment (srl295) [inline Formula/git-remote-gcrypt.rb]: @MikeMcQuaid thanks…  trying
> `    system "env prefix=#{prefix} DESTDIR=#{buildpath} ./install.sh"`

but it's not finding `rst2man` that way, and resulting in an empty installation.

File: Formula/git-remote-gcrypt.rb (hunk 2 above)
Comment (srl295) [inline Formula/git-remote-gcrypt.rb]: thanks, was able to do this… will push an  update

**Review by bfontaine — APPROVED** [review]

**Comment by woodruffw** [comment]
Any update here @srl295?

**Comment by woodruffw** [comment]
Closing due to inactivity.

**Comment by srl295** [comment]
Fair enough. I'll try to get back on this.

**Pull request closed without merging**